### PR TITLE
feat(docs): translate escape hatches

### PR DESCRIPTION
**Pull request by rayatn1011**

### CODE REVIEW

File: src/content/learn/escape-hatches.md [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
-* [How to prevent some values from re-triggering Effects](/learn/separating-events-from-effects)
-* [How to make your Effect re-run less often](/learn/removing-effect-dependencies)
-* [How to share logic between components](/learn/reusing-logic-with-custom-hooks)
+* [如何「記住」資訊而不觸發 re-rendering](/learn/referencing-values-with-refs)
+* [如何存取由 React 管理的 DOM element](/learn/manipulating-the-dom-with-refs)
+* [如何使 component 與外部系統同步](/learn/synchronizing-with-effects)
+* [如何從 component 中移除不必要的 Effect](/learn/you-might-not-need-an-effect)
+* [Effect 和 component 的生命週期有什麼區別](/learn/lifecycle-of-reactive-effects)
+* [如何防止某些數值重新觸發 Effect](/learn/separating-events-from-effects)
+* [如何降低 Effect 重新運行的頻率](/learn/removing-effect-dependencies)
```
Comment (neighborhood999) [inline src/content/learn/escape-hatches.md:19]: How about `執行`？
```suggestion
* [如何降低 Effect 重新運行的頻率](/learn/removing-effect-dependencies)
```

File: src/content/learn/escape-hatches.md [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
+* [如何在不同 component 中共享邏輯](/learn/reusing-logic-with-custom-hooks)
 
 </YouWillLearn>
 
-## Referencing values with refs {/*referencing-values-with-refs*/}
+## 藉由 Ref 參照數值 {/*referencing-values-with-refs*/}
 
-When you want a component to "remember" some information, but you don't want that information to [trigger new renders](/learn/render-and-commit), you can use a *ref*:
+當你想要在一個 component「記住」一些資訊，但是不想要讓資訊 [觸發新的 render](/learn/render-and-commit)，可以使用 *ref* ：
```
Comment (neighborhood999) [inline src/content/learn/escape-hatches.md:26]: 中文和中文之間不需要空格，下面其他部分也是😆

```suggestion
當你想要在一個 component「記住」一些資訊，但是不想要讓資訊[觸發新的 render](/learn/render-and-commit)，可以使用 *ref* ：
```

File: src/content/learn/escape-hatches.md [hunk 3]
```diff
@@ -54,17 +54,17 @@ export default function Counter() {
 
 </Sandpack>
 
-A ref is like a secret pocket of your component that React doesn't track. For example, you can use refs to store [timeout IDs](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#return_value), [DOM elements](https://developer.mozilla.org/en-US/docs/Web/API/Element), and other objects that don't impact the component's rendering output.
+Ref 就像 component 的秘密口袋，不會被 React 追蹤。例如，你可以使用 ref 來儲存 [逾時的 ID](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#return_value)、[DOM element](https://developer.mozilla.org/zh-TW/docs/Web/API/Element)，以及其他不影響 component rendering 輸出的 object。
```
Comment (neighborhood999) [inline src/content/learn/escape-hatches.md:57]:
```suggestion
Ref 就像 component 的秘密口袋，不會被 React 追蹤。例如，你可以使用 ref 來儲存[逾時的 ID](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#return_value)、[DOM element](https://developer.mozilla.org/zh-TW/docs/Web/API/Element)，以及其他不影響 component rendering 輸出的 object。
```

File: src/content/learn/escape-hatches.md [hunk 4]
```diff
@@ -93,15 +93,15 @@ export default function Form() {
 
 <LearnMore path="/learn/manipulating-the-dom-with-refs">
 
-Read **[Manipulating the DOM with Refs](/learn/manipulating-the-dom-with-refs)** to learn how to access DOM elements managed by React.
+閱讀 **[藉由 ref 操縱 DOM](/learn/manipulating-the-dom-with-refs)** 來學習如何存取由 React 管理的 DOM element。
 
 </LearnMore>
 
-## Synchronizing with Effects {/*synchronizing-with-effects*/}
+## 藉由 Effect 同步化 {/*synchronizing-with-effects*/}
```
Comment (neighborhood999) [inline src/content/learn/escape-hatches.md:100]: We can simply use `同步`？

File: src/content/learn/escape-hatches.md [hunk 5]
```diff
@@ -145,7 +145,7 @@ video { width: 250px; }
 
 </Sandpack>
 
-Many Effects also "clean up" after themselves. For example, an Effect that sets up a connection to a chat server should return a *cleanup function* that tells React how to disconnect your component from that server:
+許多 Effect 還需要「清除」自己。例如，一個與聊天伺服器建立連線的 Effect，應該 return 一個 *cleanup function * 來告訴 React 該 component 要如何與伺服器斷開：
```
Comment (neighborhood999) [inline src/content/learn/escape-hatches.md:148]:
```suggestion
許多 Effect 還需要「清除」自己。例如，一個與聊天伺服器建立連線的 Effect，應該回傳一個 *cleanup function * 來告訴 React 該 component 要如何與伺服器斷開：
```

File: src/content/learn/escape-hatches.md [hunk 6]
```diff
@@ -302,25 +302,25 @@ button { margin-left: 10px; }
...
-This section describes an **experimental API that has not yet been released** in a stable version of React.
+這個章節會介紹一個 **實驗性 API，它還沒有被發布** 在 React 的正式版本。
 
 </Wip>
 
-Event handlers only re-run when you perform the same interaction again. Unlike event handlers, Effects re-synchronize if any of the values they read, like props or state, are different than during last render. Sometimes, you want a mix of both behaviors: an Effect that re-runs in response to some values but not others.
+Event handler 只在你做了相同的互動時才重新執行。與 event handler 不同，假設 Effect 讀取的任何值，像是 prop 或 state，與上一次 render 的不一樣， Effect 就會重新執行同步。有些時候，你想要混合兩種行為：Effect 的重新執行只對某些值反應，而其他的值不會。
 
-All code inside Effects is *reactive.* It will run again if some reactive value it reads has changed due to a re-render. For example, this Effect will re-connect to the chat if either `roomId` or `theme` have changed:
+所有在 Effect 中的程式碼都是 *反應性* 的。假如某些它讀取的反應性的值，在 re-render 時發生變化，它就會再運行一次。例如：假如 `roomId` 或 `theme` 之一已經改變，那麼這個 Effect 將會重新連接到聊天室。
```
Comment (neighborhood999) [inline src/content/learn/escape-hatches.md:323]:
```suggestion
所有在 Effect 中的程式碼都是 *反應性* 的。假如某些它讀取的反應性的值，在 re-render 時發生變化，它就會再執行一次。例如：假如 `roomId` 或 `theme` 之一已經改變，那麼這個 Effect 將會重新連接到聊天室。
```

File: src/content/learn/escape-hatches.md [hunk 7]
```diff
@@ -581,19 +581,19 @@ label { display: block; margin-top: 10px; }
...
-Read **[Separating Events from Effects](/learn/separating-events-from-effects)** to learn how to prevent some values from re-triggering Effects.
+閱讀 **[從 Effect 分離事件](/learn/separating-events-from-effects)** 來學習如何防止某些值重新觸發 Effect。
 
 </LearnMore>
 
-## Removing Effect dependencies {/*removing-effect-dependencies*/}
+## 移除 Effect 的依賴 {/*removing-effect-dependencies*/}
 
-When you write an Effect, the linter will verify that you've included every reactive value (like props and state) that the Effect reads in the list of your Effect's dependencies. This ensures that your Effect remains synchronized with the latest props and state of your component. Unnecessary dependencies may cause your Effect to run too often, or even create an infinite loop. The way you remove them depends on the case.
+當你寫了一個 Effect，linter 將會驗證你是否已經將所有反應性的值（像是 prop 和 State）包含在 Effect 的依賴列表內。這確保了 Effect 會與 component 中最新的 prop 和 state 保持同步。不必要的依賴可能會導致你的 Effect 太過頻繁的執行，甚至會創建一個無窮迴圈。移除他們的方法取決於具體情況。
```
Comment (neighborhood999) [inline src/content/learn/escape-hatches.md:594]:
```suggestion
當你寫了一個 Effect，linter 將會驗證你是否已經將所有反應性的值（像是 prop 和 state）包含在 Effect 的依賴列表內。這確保了 Effect 會與 component 中最新的 prop 和 state 保持同步。不必要的依賴可能會導致你的 Effect 太過頻繁的執行，甚至會建立一個無窮迴圈。移除他們的方法取決於具體情況。
```

File: src/content/learn/escape-hatches.md [hunk 8]
```diff
@@ -581,19 +581,19 @@ label { display: block; margin-top: 10px; }
...
 
-For example, this Effect depends on the `options` object which gets re-created every time you edit the input:
+例如，這個 Effect 依賴於 `options` object，會導致在你每次編輯 input 時都被重新創建：
```
Comment (neighborhood999) [inline src/content/learn/escape-hatches.md:596]:
```suggestion
例如，這個 Effect 依賴於 `options` object，會導致在你每次編輯 input 時都被重新建立：
```

File: src/content/learn/escape-hatches.md [hunk 9]
```diff
@@ -668,7 +668,7 @@ button { margin-left: 10px; }
 
 </Sandpack>
 
-You don't want the chat to re-connect every time you start typing a message in that chat. To fix this problem, move creation of the `options` object inside the Effect so that the Effect only depends on the `roomId` string:
+你不會希望每次你開始打字聊天時，聊天室都要重新連接。為了修復這個問題，將 `options` object 的創建移到 Effect 內，這樣 Effect 就只依賴於 `roomId` string：
```
Comment (neighborhood999) [inline src/content/learn/escape-hatches.md:671]:
```suggestion
你不會希望每次你開始打字聊天時，聊天室都要重新連接。為了修復這個問題，將 `options` object 的建立移到 Effect 內，這樣 Effect 就只依賴於 `roomId` string：
```

File: src/content/learn/escape-hatches.md [hunk 10]
```diff
@@ -742,19 +742,19 @@ button { margin-left: 10px; }
...
-Read **[Removing Effect Dependencies](/learn/removing-effect-dependencies)** to learn how to make your Effect re-run less often.
+閱讀 **[移除 Effect 的依賴](/learn/removing-effect-dependencies)** 來學習如何讓你的 Effect 重新執行的頻率降低。
 
 </LearnMore>
 
-## Reusing logic with custom Hooks {/*reusing-logic-with-custom-hooks*/}
+## 藉由 custom Hook 複用邏輯 {/*reusing-logic-with-custom-hooks*/}
 
-React comes with built-in Hooks like `useState`, `useContext`, and `useEffect`. Sometimes, you’ll wish that there was a Hook for some more specific purpose: for example, to fetch data, to keep track of whether the user is online, or to connect to a chat room. To do this, you can create your own Hooks for your application's needs.
+React 有內建的 Hook 像是 `useState`、`useContext`，和 `useEffect`。有些時候，你會想要有一個更具特定目的 Hook：例如，擷取資料、持續追蹤使用者是否在線上，或者連線到一個聊天室。要做到這點，可以視應用程式的需求創建屬於自己的 Hook。
```
Comment (neighborhood999) [inline src/content/learn/escape-hatches.md:755]:
```suggestion
React 有內建的 Hook 像是 `useState`、`useContext`，和 `useEffect`。有些時候，你會想要有一個更具特定目的 Hook：例如，擷取資料、持續追蹤使用者是否在線上，或者連線到一個聊天室。要做到這點，可以視應用程式的需求建立屬於自己的 Hook。
```

File: src/content/learn/escape-hatches.md [hunk 11]
```diff
@@ -742,19 +742,19 @@ button { margin-left: 10px; }
...
 
-In this example, the `usePointerPosition` custom Hook tracks the cursor position, while `useDelayedValue` custom Hook returns a value that's "lagging behind" the value you passed by a certain number of milliseconds. Move the cursor over the sandbox preview area to see a moving trail of dots following the cursor:
+在這個範例中， `usePointerPosition`  custom Hook 追蹤了游標的位置，而 `useDelayedValue` custom Hook return 一個相對於你傳入的值有一定毫秒數延遲的值。移動游標到沙箱預覽區域，可以看到游標後有一連串在移動的點：
```
Comment (neighborhood999) [inline src/content/learn/escape-hatches.md:757]:
```suggestion
在這個範例中，`usePointerPosition` custom Hook 追蹤了游標的位置，而 `useDelayedValue` custom Hook 回傳一個相對於你傳入的值有一定毫秒數延遲的值。移動游標到沙箱預覽區域，可以看到游標後有一連串在移動的點：
```

File: src/content/learn/escape-hatches.md [hunk 12]
```diff
@@ -835,14 +835,14 @@ body { min-height: 300px; }
 
 </Sandpack>
 
-You can create custom Hooks, compose them together, pass data between them, and reuse them between components. As your app grows, you will write fewer Effects by hand because you'll be able to reuse custom Hooks you already wrote. There are also many excellent custom Hooks maintained by the React community.
+你可以創建 custom Hook，將它們組合在一起，在他們之間傳遞資料，並且在不同的 component 複用。隨著你的應用逐漸成長，你將會手寫更少的 Effect，因為你能夠複用你已經寫好的 custom Gook。而且還有很多優秀的 custom Hook 由 React 社群維護。
```
Comment (neighborhood999) [inline src/content/learn/escape-hatches.md:838]:
```suggestion
你可以建立 custom Hook，將它們組合在一起，在他們之間傳遞資料，並且在不同的 component 複用。隨著你的應用逐漸成長，你將會手寫更少的 Effect，因為你能夠複用你已經寫好的 custom Hook。而且還有很多優秀的 custom Hook 由 React 社群維護。
```

**Comment by rayatn1011** [comment]
Hi @neighborhood999, thank you for your valuable suggestion, please take a look again.

**Review by neighborhood999 — APPROVED** [review]

**Pull request merged**